core/store/src/trie/state_parts.rs

```diff
@@ -972,6 +972,78 @@ mod tests {
         }
     }
 
+    #[test]
```

Comment:
Add a comment about what this test is checking.

Reply:
+ looks like it shares a lot of logic with `test_combine_state_parts`, can we deduplicate it?